src/current/v23.1/sso-sql.md

```diff
@@ -1,158 +1,257 @@
 
-{{site.data.alerts.callout_success}}
-This example uses [`cockroach sql`]({% link {{ page.version.version }}/cockroach-sql.md %}), but you can use any SQL client that supports sufficiently long passwords.
-{{site.data.alerts.end}}
+    By default, your cluster's configuration will contain the {{ site.data.products.db }}'s own public key, allowing {{ site.data.products.db }} to serve as an IdP. When modifying this cluster setting, you must include the {{ site.data.products.db }} public key in the key set. Failing to do so will lock out console SSO users and prevent maintenance access by the {{ site.data.products.db }} managed service, leading to unintended consequences.
+
+    {{site.data.alerts.callout_success}}
+    {{ site.data.products.db }}'s IdP configuration can be viewed publicly at: [`https://cockroachlabs.cloud/.well-known/openid-configuration`](https://cockroachlabs.cloud/.well-known/openid-configuration).
+    The link specified for `jwks_uri` provides the IdP's public signing key. For {{ site.data.products.db }} this is `https://cockroachlabs.cloud/oauth2/keys`
+    {{site.data.alerts.end}}
+
+    For example:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    SET CLUSTER SETTING server.jwt_authentication.jwks = '{
+      "keys": [
+        {
+          "alg": "RS256",
+          "kty": "RSA",
+          "kid": "85ba9313fd7a7d4afa84884abcc8403004363180",
+          "e": "AQAB",
+          "use": "sig",
+          "n": "pP-rCe4jkKX6mq8yP1GcBZcxJzmxKWicHHor1S3Q49u6Oe-bQsk5NsK5mdR7Y7liGV9n0ikXSM42dYKQdxbhKA-7--fFon5isJoHr4fIwL2CCwVm5QWlK37q6PiH2_F1M0hRorHfkCb4nI56ZvfygvuOH4LIS82OzIgmsYbeEfwDRpeMSxWKwlpa3pX3GZ6jG7FgzJGBvmBkagpgsa2JZdyU4gEGMOkHdSzi5Ii-6RGfFLhhI1OMxC9P2JaU5yjMN2pikfFIq_dbpm75yNUGpWJNVywtrlNvvJfA74UMN_lVCAaSR0A03BUMg6ljB65gFllpKF224uWBA8tpjngwKQ"
+        },
+        {
+          "alg": "RS256",
+          "n": "8ASqcDDTYmFx53uAkWlmqx8qIx0WQacaswAAb7xGx9XL2T71VPtKKgPqoimXzj4fXT2F3opIgOQgGKxkxw2QrAAoejdIud5URrZLRponmbMQO3erG5sd8PC29rjekX8_hvX_AMT3zU8JLKVLFR0xJmFdmIomnarpEvWKmG0SNvbuVveprsve0n1W35uodcRe417vlTNnH9j8fesQekvFf8tIWbHouXDg3B1km4gqZBbQ_MWvGriGBY5sfr7A2d0iNe89Aje7pz1RFLUFOu-u_NyD6RwL6qo4_yetAYIzm02a2KAAq03YPs2LHMVQmkh1LtyeuA6bvf9146cAFY4BCQ",
+          "use": "sig",
+          "e": "AQAB",
+          "kty": "RSA",
+          "kid": "05150a1320b9395b05716877376928509bab44ac"
+        }
+      ]
+    }';
+    ~~~
+
+1. Configure token generation.
 
 {% include_cached copy-clipboard.html %}
 ~~~shell
-cockroach sql --url "postgresql://{SQL_USERNAME}:{JWT_TOKEN}@{CLUSTER_HOST}:26257?options=--crdb:jwt_auth_enabled=true" --certs-dir={CLUSTER_CERT_DIR}
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.enabled = true;
+SET CLUSTER SETTING server.oidc_authentication.generate_cluster_sso_token.use_token = id_token;
```

Comment:
I think this section needs more help. In order for this feature to work DB Console SSO and cluster SSO must both be configured with the same IDP. Also it might be saying that the use_token field is either `id_token` or `access_token` and what the difference means.

Reply:
thanks

Reply:
I need more help spelling this out please.

Reply:
Which part? Would it be worth grabbing some time to talk synchronously?

Reply:
We could. Just updated the text with '...' indicating where I need help. basically where you indicated.

Reply:
Hopefully this lies somewhere along the intersection of accurate and helpful. it's a tricky case.